mammoth/utils/optimizers.py

```diff
@@ -223,27 +202,17 @@ def zero_grad(self):
         for name in self.optimizers:
             self.optimizers[name].zero_grad()
 
-    def step(self, grad_scaler=None):
-        """Step through all the suboptimizers"""
+    def managed_step(self, gradient_syncs, grad_scaler=None):
```

Comment:
Here we also use the term `managed` with the same meaning.
The caller must supply gradient syncs (a sequence of ` DistributedComponentGradientSync`. Should add a type annotation) which determine whether or not to step each suboptimizer.